cimr/processor/yamler.py

```diff
@@ -0,0 +1,400 @@
+        reversekeys = {
+            v: k for k, v in columnset.items() if v != 'na'
+        }
+        # get a list of new submitted_data files to process
+        infile = self.file_link.split('/')[-1]
+        outfile = self.yaml_data['data_file']['output_name']
+        downloaded_data = pandas.read_csv(
+            self.outdir + infile, 
+            sep='\t',
+            header=0
+        )
+        # if 'output_name' in self.yaml_data['data_file']:
+        #     if self.yaml_data['data_file']['output_name'] is not 'na':
+        #         outfile = 
+        logging.info(f' renaming columns based on provided info')
+        renamed_data = downloaded_data.rename(reversekeys, axis=1)
+        self.outfile_path = self.outdir + outfile + '.txt.gz'
+        logging.info(f' writing renamed dataset to a file: {self.outfile_path}')
+        renamed_data.to_csv(
+            self.outfile_path,
+            sep='\t',
+            header=True,
+            index=False,
+            na_rep='NA',
+            compression='gzip'
+        )
+        if os.path.isfile(self.outfile_path):
+            self.make_archive_dir()
+            os.rename(
+                self.outdir + infile,
+                self.archive_dir + infile
+            )
+            logging.info(f' moving downloaded file to the archive')
+        # variant_id = columnset['variant_id']
+        # variant_chrom = columnset['variant_chrom']
+        # variant_pos = columnset['variant_pos']
+        # rsnum = columnset['rsnum']
+        # ref = columnset['ref']
+        # alt = columnset['alt']
+        # build = columnset['build']
+        # effect_allele = columnset['effect_allele']
+        # non_effect_allele = columnset['non_effect_allele']
+        # inc_allele = columnset['inc_allele']
+        # inc_afrq = columnset['inc_afrq']
+        # effect_size = columnset['effect_size']
+        # standard_error = columnset['standard_error']
+        # statistic = columnset['statistic']
+        # pvalue = columnset['pvalue']
+        # feature_id = columnset['feature_id']
+        # feature_chrom = columnset['feature_chrom']
+        # feature_start = columnset['feature_start']
+        # feature_stop = columnset['feature_stop']
+        # imputation_status = columnset['imputation_status']
+        # frequency = columnset['frequency']
+        # tss_distance = columnset['tss_distance']
+        # ma_samples = columnset['ma_samples']
+        # maf = columnset['maf']
+        # comment_0 = columnset['comment_0']
+
+
+    def check_data_file(self):
+        """Standard set of Yamler functions to check information on the
+        contributed data file for ci cimr processing.
+        """
+        self.set_data_type()
+        self.check_defined()
+        self.format_data()
+
+
+if __name__ == '__main__':
+
+    logging.basicConfig(level='DEBUG')
+
+    if len(sys.argv) == 1:
+        yaml_files = find_yaml_in_dir()
+    else:
+        yaml_files = pathlib.Path(sys.argv[1])
```

Comment:
According to line 389, `yaml_files` must be a list of filenames, not a regular string. So this line should be modified to:
```python
yaml_files = [pathlib.Path(sys.argv[1]), ]
```
Otherwise line 389 will iterate the letters in a single file.

Reply:
good catch! thank you!